lib/debugging.js

```diff
@@ -1,18 +1,33 @@
 /* @flow weak */
 "use strict";
-module.exports = {
+
+var debug = require("debug");
+
+function setLoggersHelper(debugFns, loggerFactory) {
+  Object.keys(debugFns).filter(function(key) {
+    return (key.substr(0, 2) !== "__");
+  }).forEach(function(key) {
+    if (debugFns[key] instanceof Function) {
+      debugFns[key] = loggerFactory(debugFns[key].namespace);
+    }
+    return setLoggersHelper(debugFns[key], loggerFactory);
+  });
+}
+
+var debugging = module.exports = {
   handler: {
-    search: require("debug")("jsonApi:handler:search"),
-    find: require("debug")("jsonApi:handler:find"),
-    create: require("debug")("jsonApi:handler:create"),
-    update: require("debug")("jsonApi:handler:update"),
-    delete: require("debug")("jsonApi:handler:delete")
+    search: debug("jsonApi:handler:search"),
+    find: debug("jsonApi:handler:find"),
+    create: debug("jsonApi:handler:create"),
+    update: debug("jsonApi:handler:update"),
+    delete: debug("jsonApi:handler:delete")
   },
-  include: require("debug")("jsonApi:include"),
-  filter: require("debug")("jsonApi:filter"),
-  validationInput: require("debug")("jsonApi:validation:input"),
-  validationOutput: require("debug")("jsonApi:validation:output"),
-  validationError: require("debug")("jsonApi:validation:error"),
-  errors: require("debug")("jsonApi:errors"),
-  requestCounter: require("debug")("jsonApi:requestCounter")
+  include: debug("jsonApi:include"),
+  filter: debug("jsonApi:filter"),
+  validationInput: debug("jsonApi:validation:input"),
+  validationOutput: debug("jsonApi:validation:output"),
+  validationError: debug("jsonApi:validation:error"),
+  errors: debug("jsonApi:errors"),
+  requestCounter: debug("jsonApi:requestCounter"),
+  __setLoggers: setLoggersHelper.bind(null, debugging)
```

Comment:
Is `setLoggers` a good choice of function name? Should it be more like `.__redirectOutputTo(winston)` or similar?